### DISCUSSION TITLE
Creating a Select Component with Submenus 🌟

### DISCUSSION THREAD

**Discussion by pavelbe** [shadcn-ui/ui#2793] [Q&A]
Hello, shadcn-ui community! 👋

I'm currently working with the Select component within the shadcn-ui library and am interested in enhancing its functionality to include submenus. Specifically, I'm looking to create a hierarchical structure like "Category > Menu > Submenu". This structure would allow for a more organized and user-friendly selection process, especially in cases where the list of options is extensive.

Has anyone in this community tackled a similar challenge or implemented a Select component with submenus in shadcn-ui? If so, I would greatly appreciate it if you could share your experience, insights, or even a code snippet that demonstrates how you achieved this functionality. Any guidance or examples would be incredibly helpful as I navigate this implementation.

Thank you in advance for your help and for sharing your knowledge! 🙏

**Comment by maiconcarraro** [comment]
Hey @pavelbe did you check the [Dropdown](https://ui.shadcn.com/docs/components/dropdown-menu)?
![image](https://github.com/shadcn-ui/ui/assets/13419087/1f2e9d5b-4349-4d31-9700-8f295e7c3000)
